pandas/core/indexes/multi.py

```diff
@@ -2628,13 +2628,25 @@ def _partial_tup_index(self, tup, side="left"):
                     loc -= 1
                 return start + section.searchsorted(loc, side=side)
 
-            idx = lev.get_loc(lab)
+            idx = self._get_loc_single_index_wrapper(lev, lab)
             if k < n - 1:
                 end = start + section.searchsorted(idx, side="right")
                 start = start + section.searchsorted(idx, side="left")
             else:
                 return start + section.searchsorted(idx, side=side)
 
+    def _get_loc_single_index_wrapper(self, level_index, key):
+        """
+        single index get_loc wrapper function for multi index.
+        Depending on index type, Handling NA value is different using get_loc.
+        But in multi index, NA values is denoted as missing by -1.
+        If key is NA value, location of index unify as -1.
+        """
+        if not is_list_like(key) and isna(key):
```

Comment:
how is this *every* list-like? isn't it supposed to be a scalar?

Reply:
@jreback 
I change into check scalar or not. In `PeriodIndex`, all key except when key is tuple, call single index get_loc. For a NA value type check, it is inevitable.